pkg/download/api.go

```diff
@@ -36,6 +37,11 @@ func (DownloadClientFactory) NewDownloadClient(config model.DownloadClient) (Dow
 	case "transmission":
 		// TODO: Replace default http client with stored configurations
 		return NewTransmissionClient(http.DefaultClient, config.Scheme, config.Host, int(config.Port)), nil
+	case "sabnzbd":
+		if config.APIKey == nil {
+			return nil, errors.New("missing api key")
+		}
+		return NewSabnzbdClient(http.DefaultClient, config.Scheme, config.Host, *config.APIKey), nil
```

Comment:
looking at this again, I wonder if we should just pass the entire config, and then have the different implementations handle creating their own http client + whatever else based on those configs

Reply:
I do generally like the New being explicit which fields are used. Especially since you're switching on implementation either way

Reply:
works for me